src/test/crimson/test_messenger.cc

```diff
@@ -1574,7 +1583,8 @@ class FailoverTestPeer : public Dispatcher {
     cmd_msgr->set_auth_client(&dummy_auth);
     cmd_msgr->set_auth_server(&dummy_auth);
     return cmd_msgr->bind(entity_addrvec_t{cmd_peer_addr}).then([this] {
-      return cmd_msgr->start(this);
+      cmd_msgr->add_dispatcher(this);
```

Comment:
Even more cases where having add_dispatcher() return *this for chaining will simplify the code